### DISCUSSION TITLE
Database error: ER_ACCESS_DENIED_NO_PASSWORD_ERROR

### DISCUSSION THREAD

**Discussion by zougui1** [adonisjs/core#2275] [Help] [answered]
I can't connect to the database despite providing the correct password.

I get this error
``ER_ACCESS_DENIED_NO_PASSWORD_ERROR: Access denied for user 'zougui'@'localhost'`` 
It always say there is no password even though it is defined, and doing a ``console.log`` in the config file prints the correct password so I don't understand what is wrong.
And I am sure the password works, I can connect to mysql from the terminal and have done it with other ORMs for the same user/password within the past 2 months.

Here is my .env file
```
DB_CONNECTION=mysql
MYSQL_HOST=localhost
MYSQL_PORT=3306
MYSQL_USER=zougui
MYSQL_PASSWORD=some_password
MYSQL_DB_NAME=music_manager
```
The config of the database
```ts
console.log('MYSQL_PASSWORD', Env.get('MYSQL_PASSWORD')); // prints 'some_password'

const databaseConfig: DatabaseConfig & { orm: Partial<OrmConfig> } = {
  connection: Env.get('DB_CONNECTION'),

  connections: {
    mysql: {
      client: 'mysql',
      connection: {
        host: Env.get('MYSQL_HOST'),
        port: Env.get('MYSQL_PORT'),
        user: Env.get('MYSQL_USER'),
        password: Env.get('MYSQL_PASSWORD'),
        database: Env.get('MYSQL_DB_NAME'),
      },
      healthCheck: true,
      debug: true,
    },
  },
}
```

I'm on Arch Linux and use MariaDB 10.5.8 if that matters.

Versions:
Node: 14.15.4
@adonijs/core: 5.0.4-preview-rc-2.1
@adonisjs/lucid: 10.0.0
mysql: 2.18.1

EDIT:
I created a new user in the DB and with this one it works. So at least I can keep working. But I wonder why with the other user it doesn't work? 
Also I think there is an error with the errors being thrown.
With the user that doesn't work, no matter if I give it a password or not in the .env file, Lucid throws an ER_ACCESS_DENIED_NO_PASSWORD_ERROR error. 
With the new user I created, if I don't give it a password in the .env file, Lucid throws an ER_ACCESS_DENIED_ERROR error, not an ER_ACCESS_DENIED_NO_PASSWORD_ERROR.

**Comment by thetutlage** [comment, marked as answer]
These errors comes directly from the underlying mysql driver. Have to tried making a connection directly using the `mysql` package. If yes, can you please share code for that?

**Reply by zougui1** [reply]
Ok thanks I tried, it threw on error specifying it was the connection to the database I was trying to connect to. So I checked the databases from the terminal and I noticed I forgot to add the rights of the database to the user. Though the error was different from what I had yesterday so I retried with Lucid ORM without the rights and the error was the same as the driver, like you said, though it is _different_ from the error I had yesterday, and here again I don't know why.
In any case, the problem is solved.
